modules/runx/runx.py

```diff
@@ -50,10 +48,19 @@ def process(self, module_idx: int) -> None:
         logger.debug("Checking for runx api key secret")
         current_api_key = self.fetch_secret()
         if current_api_key is None:
-            self.set_secret()
+            current_api_key = self.set_secret()
         else:
             self.fetch_jwt(current_api_key)
 
+        self.module.data["api_key"] = current_api_key
+        k8s_modules = self.layer.get_module_by_type("k8s-cluster", module_idx)
+        if len(k8s_modules) == 0 and self.module.data.get("deploy_k8s_listener") is True:
```

Comment:
Since `deploy_k8s_listener` is true by default. this error could be confusing as a user might not have explicitly set this. Let's just automagically set that to false here instead of an error

Reply:
no, because that default does not get applied until later. self.module.data.get("deploy_k8s_listener") returns None if the user has not set it